phpBB/includes/acp/acp_forums.php

```diff
@@ -986,10 +986,30 @@ function update_forum_data(&$forum_data_ary)
 			$errors[] = $user->lang['FORUM_NAME_EMPTY'];
 		}
 
-		// No Emojis
+		/**
+		 * Replace Emojis and other 4bit UTF-8 chars not allowed by MySql to NCR.
+		 * Using their Numeric Character Reference's Hexadecimal notation.
+		 * Doesn't interfere with Japanese or Cyrillic etc.
+		 *
+		 * @see https://www.w3.org/TR/xml11/
+		 * @see https://www.opentag.com/xfaq_charrep.htm
+		 */
+		if (preg_match_all('/[\x{10000}-\x{10FFFF}]/u', $forum_data_ary['forum_name'], $matches))
```

Comment:
Instead of running as many `str_replace` as there are replacements, you should be able to run a single replacement with the `utf8_encode_ncr_callback`.

```php
function utf8_encode_sp($text)
{
	return preg_replace_callback('/[\\xF0-\\xF4].../', 'utf8_encode_ncr_callback', $text);
}
```

Then the whole block becomes a single assignment:
```php
$forum_data_ary['forum_name'] = utf8_encode_sp($forum_data_ary['forum_name']);
```

Reply:
I don't see the need to add a new function IMHO.

Let's not forget that we are in ACP, the admin is choosing a name for a forum that most likely will never change, if he ever did I doubt that it is a constant operation over time.

Why do I have to pass under the press a string if there is no need to do it? :thinking: 
Your regex seems to me to evaluate the same range of reserved characters (SMP) as mine, the native function receives and evaluates only the characters extracted from my regex. 

I admit I thought about a solution like yours during the developmment of this idea, I chose to go this way though.

If, and could be possible indeed, I am missing something pls let me know, thanks @JoshyPHP .

Reply:
It's a good idea to have a function for it so you can use it in [your other PR](https://github.com/phpbb/phpbb/pull/5685/files#diff-2f48d04a5ad1975b5751ca778161d39b) instead of having the same code twice or more. It's got fewer lines and a lower cyclomatic complexity.

The regexp is functionally the same, it's just cheaper to operate on bytes than on codepoints.

Reply:
I am bound to open 2 more PRs about Emojis, I will see what a do after some test. Thx Joshy.

Reply:
I also think adding a function for this makes more sense. If the counter argument is only "oh no, we have to add a function", then I'd definitely go for it. :tada:

Reply:
Okay. I will add a function, no probs.